UEBPrep/app/models/card.rb

```diff
@@ -0,0 +1,28 @@
+class Card < ActiveRecord::Base
+  validates :content_type_id, presence: true
+  validates :content_id, presence: true
+  validates :created_by, presence: true
+  validates_uniqueness_of :content_id, :scope => :content_type_id
+
+  attr_accessor :content
+
+  after_initialize :get_content
+  after_find       :get_content
+  before_destroy   :destroy_content
+
+  private
+    # I hate everything about needing to do this ...
+    # something not quite right with my schema
+    def get_content
+      case self.content_type_id
+        when 1
+          @content = TextContent.find(self.content_id)
+        when 2
+          @content = TitledCardContent.find(self.content_id)
+      end
+    end
+
+    def destroy_content
+      content.destroy
```

Comment:
There is a active record callback we could use, like after_destroy dependent destroy something-basically meaning after you destroy this model go delete everything that was linked to it.